test/byContract/IexecPoco/IexecPoco1.test.ts

```diff
@@ -204,10 +231,219 @@ describe('IexecPoco1', () => {
+            );
+        });
+
+        it('[TEE] Should fail when category is unknown', async () => {
+            const categoriesCount = (await iexecPoco.countCategory()).toNumber();
+            orders.requester.category = categoriesCount + 1;
+            orders.workerpool.category = categoriesCount + 1;
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x01',
+            );
+        });
+
+        it('[TEE] Should fail when requested trust is above workerpool trust', async () => {
+            orders.requester.trust = 123;
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x02',
+            );
+        });
+
+        it('[TEE] Should fail when app max price is less than app price', async () => {
+            orders.requester.appmaxprice = Number(orders.app.appprice) - 1;
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x03',
+            );
+        });
+
+        it('[TEE] Should fail when dataset max price is less than dataset price', async () => {
+            orders.requester.datasetmaxprice = Number(orders.dataset.datasetprice) - 1;
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x04',
+            );
+        });
+
+        it('[TEE] Should fail when workerpool max price is less than workerpool price', async () => {
+            orders.requester.workerpoolmaxprice = Number(orders.workerpool.workerpoolprice) - 1;
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x05',
+            );
+        });
+
+        it('[TEE] Should fail when workerpool tag does not match all app, dataset and request tags', async () => {
+            orders.app.tag = '0x0000000000000000000000000000000000000000000000000000000000000001'; // 0b0001
+            orders.dataset.tag =
+                '0x0000000000000000000000000000000000000000000000000000000000000002'; // 0b0010
+            orders.requester.tag =
+                '0x0000000000000000000000000000000000000000000000000000000000000003'; // 0b0011
+            // Workerpool order is supposed to satisfy conditions of all actors.
+            // Bad tag, correct tag should be 0b0011.
+            orders.workerpool.tag =
+                '0x0000000000000000000000000000000000000000000000000000000000000004'; // 0b0100
+            // Match orders.
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x06',
+            );
+        });
+
+        it('[TEE] Should fail when the last bit of app tag does not match dataset or request tags', async () => {
```

Comment:
I think boost name for it is more appropriate since here you compare 1 bit to bytes.
Same above imo where it looks all tags must match together

Reply:
Update, LMK.